src/Parallel/TypeTraits.hpp

```diff
@@ -0,0 +1,33 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines type traits related to Charm++ types
+
+#pragma once
+
+#include <charm++.h>
+
+namespace Parallel {
+
+/// \ingroup Parallel
```

Comment:
I want to raise the question: Do we think these should be in the `TypeTraits` doxygen group? I feel fairly strongly against putting them there because these are so low-level and tied to Charm++ and parallelism that I'd rather not have them be exposed elsewhere. However, I'd like to give a place for discussion since maybe there is a good reason someone else has for adding them to the TypeTraits group.